lifecycle/lifecycle/supervisor/metrics.py

```diff
@@ -0,0 +1,39 @@
+from pathlib import Path
+
+from lifecycle.config import Config
+from lifecycle.fatman.registry import list_fatmen_registry
+from racetrack_client.log.context_error import wrap_context
+from racetrack_client.log.logs import get_logger
+from racetrack_commons.entities.dto import FatmanDto, FatmanStatus
+from racetrack_commons.plugin.loader import ensure_dir_exists
+
+logger = get_logger(__name__)
+
+
+def populate_metrics_jobs(config: Config):
+    """
+    Populate a configuration file for Prometheus to discover all the jobs
+    using `file_sd_configs` feature
+    """
+    metrics_dir = Path(config.plugins_dir) / 'metrics'
+    ensure_dir_exists(metrics_dir)
+    service_discovery_file: Path = metrics_dir / 'sd_config_jobs.yaml'
+
+    file_content: list[str] = []
+    with wrap_context('populating Prometheus configuration'):
+        for job in list_fatmen_registry(config):
+            if job.status != FatmanStatus.LOST.value:
+                file_content.append(_build_job_config(job))
```

Comment:
This means that when Fatman starts having problems, the metrics for it won't be collected for the nearest minute (or longer). I wonder how Prometheus reacts to such disappearing jobs - my assumption is it won't forget them, right? And if job reappears, it will pick it up, and collect metrics to the same job. It makes sense that's how it should work, but would be good to confirm.

Reply:
Yep, if a job disappears, Prometheus still keeps track of it, job has just null values then. When it appears again, it picks it up.

"when Fatman starts having problems" it usually has "ERROR" status and notice that Prometheus will still try to scrape from it, that's intended.  I only excluded "LOST" status as it usually means that the cluster was wiped out and there's no need to bother Prometheus.